compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt

```diff
@@ -91,6 +98,20 @@ private class BackendCodeGenerator(
     /** Generates code for [host]'s role in the function [functionDeclaration]. */
     private fun generate(functionDeclaration: FunctionDeclarationNode): FunSpec {
         val hostFunctionBuilder = FunSpec.builder(functionDeclaration.name.value.name)
+
+        for (param in functionDeclaration.parameters.filter { param -> param.protocol!!.value.hosts.contains(host) }) {
+            val paramName = context.kotlinName(param.name.value)
+            val paramType = codeGenerator.kotlinType(param.protocol!!.value, typeAnalysis.type(param))
+            if (param.isInParameter) {
+                hostFunctionBuilder.addParameter(paramName, paramType)
+            } else {
+                hostFunctionBuilder.addParameter(
+                    context.outBoxName(paramName),
+                    Out::class.asClassName().parameterizedBy(paramType)
+                )
+                hostFunctionBuilder.addStatement("val %N: %T", paramName, paramType)
```

Comment:
This may not work for all functions. For example, if an out parameter is initialized by a function call.